cdc/api/v2/validator.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/cdc/capture"
+	"github.com/pingcap/tiflow/cdc/contextutil"
+	"github.com/pingcap/tiflow/cdc/entry"
+	"github.com/pingcap/tiflow/cdc/kv"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/sink"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/security"
+	"github.com/pingcap/tiflow/pkg/txnutil/gc"
+	"github.com/pingcap/tiflow/pkg/util"
+	"github.com/pingcap/tiflow/pkg/version"
+	"github.com/tikv/client-go/v2/oracle"
+	pd "github.com/tikv/pd/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+)
+
+// VerifyCreateChangefeedConfig verifies ChangefeedConfig and
+// returns a  changefeedInfo for create a changefeed.
+func VerifyCreateChangefeedConfig(
+	ctx context.Context,
+	cfg *ChangefeedConfig,
+	capture *capture.Capture,
+) (*model.ChangeFeedInfo, error) {
+	// verify credential and pdAddrs
+	if len(cfg.PDAddrs) == 0 {
+		return nil, cerror.ErrAPIInvalidParam.GenWithStackByCause("can not create a changefeed with empty pdAddrs")
+	}
+	credential := security.Credential{
+		CAPath:   cfg.CAPath,
+		CertPath: cfg.CertPath,
+		KeyPath:  cfg.KeyPath,
+	}
+	grpcTLSOption, err := credential.ToGRPCDialOption()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(cfg.CertAllowedCN) != 0 {
+		credential.CertAllowedCN = cfg.CertAllowedCN
+	}
+
+	pdClient, err := pd.NewClientWithContext(
+		ctx, cfg.PDAddrs, credential.PDSecurityOption(),
+		pd.WithGRPCDialOptions(
+			grpcTLSOption,
+			grpc.WithBlock(),
+			grpc.WithConnectParams(grpc.ConnectParams{
+				Backoff: backoff.Config{
+					BaseDelay:  time.Second,
+					Multiplier: 1.1,
+					Jitter:     0.1,
+					MaxDelay:   3 * time.Second,
+				},
+				MinConnectTimeout: 3 * time.Second,
+			}),
+		))
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer pdClient.Close()
+
+	// verify sinkURI
+	if cfg.SinkURI == "" {
+		return nil, cerror.ErrSinkURIInvalid.GenWithStackByArgs("sink-uri is empty, can't not create a changefeed without sink-uri")
+	}
+
+	// verify changefeedID
+	if err := model.ValidateChangefeedID(cfg.ID); err != nil {
```

Comment:
if the changefeed id is not specified , we should generate an UUID ?

Reply:
sure, good catch